pkgs/development/tools/documentation/gnome-doc-utils/default.nix

```diff
@@ -12,28 +12,46 @@ python2Packages.buildPythonApplication rec {
     sha256 = "19n4x25ndzngaciiyd8dd6s2mf9gv6nv3wv27ggns2smm7zkj1nb";
   };
 
+  patches = [
+    # https://bugzilla.redhat.com/show_bug.cgi?id=438638
+    (fetchpatch {
+      url = "https://src.fedoraproject.org/rpms/gnome-doc-utils/raw/6b8908abe5af61a952db7174c5d1843708d61f1b/f/gnome-doc-utils-0.14.0-package.patch";
+      sha256 = "sha256-V2L2/30NoHY/wj3+dsombxveWRSUJb2YByOKtEgVx/0=";
+    })
+    # python3 support
+    (fetchpatch {
+      url = "https://src.fedoraproject.org/rpms/gnome-doc-utils/raw/6b8908abe5af61a952db7174c5d1843708d61f1b/f/gnome-doc-utils-0.20.10-python3.patch";
+      sha256 = "sha256-niH/Yx5H44rsRgkCZS8LWLFB9ZvuInt75zugzoVUhH0=";
+    })
+  ];
+
   nativeBuildInputs = [ intltool pkg-config libxslt.dev ];
-  buildInputs = [ libxslt ];
+  buildInputs = [ libxml2 libxslt ];
+  propagatedBuildInputs = [ python3Packages.libxml2 ];
 
   configureFlags = [ "--disable-scrollkeeper" ];
 
   preBuild = ''
-    substituteInPlace xml2po/xml2po/Makefile --replace '-e "s+^#!.*python.*+#!$(PYTHON)+"' '-e "s\"^#!.*python.*\"#!$(PYTHON)\""'
+    substituteInPlace xml2po/xml2po/Makefile \
+      --replace '-e "s+^#!.*python.*+#!$(PYTHON)+"' '-e "s\"^#!.*python.*\"#!$(PYTHON)\""'
   '';
 
-  propagatedBuildInputs = [ libxml2Python ];
+  # patchShebags didn't work on these
+  postFixup = ''
+    substituteInPlace $out/bin/gnome-doc-prepare \
+      --replace " /bin/sh" "${bash}/bin/bash"
+    substituteInPlace $out/bin/gnome-doc-tool \
+      --replace "/bin/bash" "${bash}/bin/bash"
+    # Do not propagate Python
+    rm $out/nix-support/propagated-build-inputs
```

Comment:
Why not move the python package to nativeBuildInputs instead?

Reply:
the python module is used during runtime so idk why that file is deleted. i removed that line